core/src/test/scala/org/typelevel/paiges/PaigesScalacheckTest.scala

```diff
@@ -212,50 +212,21 @@ class PaigesScalacheckTest extends FunSuite {
   }
 
   test("the left and right side of a union are the same after flattening") {
-    import Doc._
-    def okay(d: Doc): Boolean = d match {
-      case Empty | Text(_) | Line(_) => true
-      case Concat(a, b) => okay(a) && okay(b)
-      case Nest(j, d) => okay(d)
-      case Align(d) => okay(d)
-      case LazyDoc(d) => okay(d.evaluated)
-      case Union(a, b) =>
-        (a.flatten === b.flatten) && okay(a) && okay(b)
+    forAll { u: Doc.Union => assert(u.a.flatten === u.b.flatten) }
+  }
+
+  test("the left side of a union has a first line as long or longer than the right") {
+    forAll(Gen.choose(1, 200), genUnion) { (n, u) =>
+      if (u.nonEmpty) {
+        def firstLine(d: Doc) = {
+          val stream = d.renderStream(n)
+          val inits = stream.takeWhile(s => !s.contains("\n"))
+          (inits :+ stream.head.takeWhile(_ != '\n')).mkString
```

Comment:
why is the head on the last element of the list? I am confused by this.

could we instead do:
```scala
def firstLine(s: Stream[String], acc: List[String]): String =
  s match {
    case Stream.Empty => acc.reverse.mkString
    case head #:: _ if head.contains('\n') => (head.takeWhile(_ != '\n') :: acc).reverse.mkString
    case head #:: tail => firstLine(tail, head :: acc)
}
```

It seems clearer to me that `firstLine(stream, Nil)` is what we want.

Reply:
Arg, this is just my lack of familiarity with Scala. I confused Stream and Iterator, and assumed the `takeWhile` would consume the first lines, so the first element after the `takeWhile` would be the one with the newline.  There's a bug even if I was correct about that, since I assume there's a \n in the input in the first place.  Sorry for the terrible PR!

Reply:
This is done.  Thank you for the review.